kaldb/src/main/java/com/slack/kaldb/logstore/search/aggregations/AggBuilderBase.java

```diff
@@ -0,0 +1,42 @@
+package com.slack.kaldb.logstore.search.aggregations;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Provides common functionality for all aggregations. All aggregations are expected to extend this
+ * class, or another base class that extends this one (ie, ValueSourceAggBuilder).
+ */
+public abstract class AggBuilderBase implements AggBuilder {
```

Comment:
I think we can call this class `AggBuilder` and not have the AggBuilder class that we have today

And then this class can declare `getType` as an abstract method?
```
abstract String getType();
```

I tried it locally and it seems to work

Reply:
This is correct, we can simplify this as you mentioned and it will work. The reason for this approach was that it mirrors that of OpenSearch abstract class `AggregationBuilder` and the `BaseAggregationBuilder` [interface](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/search/aggregations/BaseAggregationBuilder.java). The `AggBuilder` interface exists so that we have an easy option for adding pipeline aggregation support in the future when we decide to support it.

Reply:
sgtm!